crml/staking/src/lib.rs

```diff
@@ -1035,7 +1012,13 @@ decl_module! {
 			// You're auto-bonded forever, here. We might improve this by only bonding when
 			// you actually validate/nominate and remove once you unbond __everything__.
 			<Bonded<T>>::insert(&stash, &controller);
-			<Payee<T>>::insert(&stash, payee);
+
+			// controller destination is just a custom account after all
+			if let RewardDestination::Controller = payee {
+				<Payee<T>>::insert(&stash, RewardDestination::Account(controller.clone()));
```

Comment:
Why we don't simply insert the actual account?

Reply:
We could do that. Should be in its own PR though